python/tests/conftest.py

```diff
@@ -21,20 +21,116 @@
+    # simpliest
+    with mlflow.start_run(experiment_id=experiment_id):
+        mlflow.log_param("optimizer", "Adam")
+        # Fake training loop
+        model = torch.load(resnet_model_uri)
+        artifact_path = "model"
+
+        schema = (
+            "STRUCT<boxes:ARRAY<ARRAY<float>>,"
+            "scores:ARRAY<float>,label_ids:ARRAY<int>>"
+        )
+        pre_processing = (
+            "rikai.contrib.torch.transforms."
+            "fasterrcnn_resnet50_fpn.pre_processing"
+        )
+        post_processing = (
+            "rikai.contrib.torch.transforms."
+            "fasterrcnn_resnet50_fpn.post_processing"
+        )
+        rikai.mlflow.pytorch.log_model(
+            model,  # same as vanilla mlflow
+            artifact_path,  # same as vanilla mlflow
+            schema,
+            pre_processing,
+            post_processing,
+            registered_model_name="rikai-test",  # same as vanilla mlflow
+        )
+
+    # vanilla mlflow
+    with mlflow.start_run():
+        mlflow.pytorch.log_model(
+            model, artifact_path, registered_model_name="vanilla-mlflow"
+        )
+        mlflow.set_tags(
+            {
+                "rikai.model.flavor": "pytorch",
+                "rikai.output.schema": schema,
+                "rikai.transforms.pre": pre_processing,
+                "rikai.transforms.post": post_processing,
+            }
+        )
+
+    # vanilla mlflow no tags
+    with mlflow.start_run():
+        mlflow.pytorch.log_model(
+            model,
+            artifact_path,
+            registered_model_name="vanilla-mlflow-no-tags",
+        )
+
+    # vanilla mlflow wrong tags
+    with mlflow.start_run():
+        mlflow.pytorch.log_model(
+            model,
+            artifact_path,
+            registered_model_name="vanilla-mlflow-wrong-tags",
+        )
+        mlflow.set_tags(
+            {
+                "rikai.model.flavor": "pytorch",
+                "rikai.output.schema": schema,
+                "rikai.transforms.pre": "wrong_pre",
+                "rikai.transforms.post": "wrong_post",
+            }
+        )
+    return mlflow.tracking.MlflowClient(tracking_uri), tracking_uri
+
+
+@pytest.fixture(scope="module")
+def mlflow_client(mlflow_client_with_tracking_uri):
+    return mlflow_client_with_tracking_uri[0]
+
+
+@pytest.fixture(scope="module")
+def mlflow_tracking_uri(mlflow_client_with_tracking_uri):
+    return mlflow_client_with_tracking_uri[1]
+
+
+@pytest.fixture(scope="module")
+def spark(mlflow_tracking_uri: str) -> SparkSession:
+    print(f"ml flow tracking uri for spark: ${mlflow_tracking_uri}")
     rikai_version = get_default_jar_version(use_snapshot=True)
     hadoop_version = "3.2.0"  # TODO(lei): get hadoop version
+    # Avoid reused session polluting configs
+    active_session = SparkSession.getActiveSession()
+    if active_session:
+        print("active session stopped, will restart")
```

Comment:
nit: print

Reply:
It's in test code base, so I think print is ok?